operator/pkg/gateway-api/gateway_reconcile.go

```diff
@@ -318,7 +318,8 @@ func isAttachable(_ context.Context, gw *gatewayv1.Gateway, route metav1.Object,
 func (r *gatewayReconciler) setAddressStatus(ctx context.Context, gw *gatewayv1.Gateway) error {
 	svcList := &corev1.ServiceList{}
 	if err := r.Client.List(ctx, svcList, client.MatchingLabels{
```

Comment:
wouldn't it be sufficient to just use an additional `ListOption` (`client.InNamespace(gw.GetNamespace()`) ? (the LB `Service` is always in the same namespace as the `Gateway` anyway)

Hence i don't think that it's necessary to introduce an additional label.

Reply:
good call out, I was overthinking about cross namespace like one of Ingress cases.